### ISSUE TITLE
Couldn't obtain the same performance reported in the paper

### ISSUE THREAD

**Issue by siyueyu** [fanq15/GCoNet#6]
Hi, it is a really interesting idea for co-saliency. 
I tried the training code  with your default hyper-parameters. However, i couldn't get the same result as in the paper. 
Could you please share some reasons with me?

**Comment by fanq15** [comment]
Hi, you can check the `Note` in Readme.
We observe that the results on CoCA dataset are unstable when train the model multiple times, especifically for the F-measure metric. It may be caused by the challenging images of CoCA dataset where the target objects are relative small and there are many non-target objects in a complex environment. It is also a potential research direction to obtain stable results on such challenging real-world images. We follow other CoSOD methods to report the best performance of our model. You need to train the model multiple times to obtain the best result on CoCA dataset.

**Comment by kashima71** [comment]
> Hi, you can check the `Note` in Readme.
> We observe that the results on CoCA dataset are unstable when train the model multiple times, especifically for the F-measure metric. It may be caused by the challenging images of CoCA dataset where the target objects are relative small and there are many non-target objects in a complex environment. It is also a potential research direction to obtain stable results on such challenging real-world images. We follow other CoSOD methods to report the best performance of our model. You need to train the model multiple times to obtain the best result on CoCA dataset.

Then you should report the average results (say 5 or 10 times), rather than report the "best" one after many tries.

**Comment by fanq15** [comment]
You are right. We just follow other papers.

**Comment by siyueyu** [comment]
@fanq15  I found that during training, the loss of auxiliary classifier couldn't decrease, nearly remain unchanged. Why would this happen?

**Comment by fanq15** [comment]
@siyueyu In our initial experiments, we did not try many times.

**Comment by fanq15** [comment]
> > Hi, you can check the `Note` in Readme.
> > We observe that the results on CoCA dataset are unstable when train the model multiple times, especifically for the F-measure metric. It may be caused by the challenging images of CoCA dataset where the target objects are relative small and there are many non-target objects in a complex environment. It is also a potential research direction to obtain stable results on such challenging real-world images. We follow other CoSOD methods to report the best performance of our model. You need to train the model multiple times to obtain the best result on CoCA dataset.
> 
> Then you should report the average results (say 5 or 10 times), rather than report the "best" one after many tries.

In our initial experiments, we did not try many times. We just obtain this performance and report it. I did not realize the unstable problem on CoCA dataset at that time. Actually, even report the averaged results, our results are still much better than other methods on CoCA dataset. If I want to report the "best" result after many tries as you said. I should do it on Cosal2015 and CoSOD3k datasets for better performance gains. However, I have to say, I should run many tries to estimate the performance stability and report a stable result. Sorry about that. If you are interested, we can discuss it more.

**Comment by fanq15** [comment]
@siyueyu The loss of auxiliary classifier decreases. Its loss at the first epoch is around 17-20 and at the 8-th epoch, the loss decreases to 12-15.
